ui/apps/platform/src/services/ReportsService.ts

```diff
@@ -225,3 +223,23 @@ export function deleteDownloadableReport(reportId: string) {
         return response.data;
     });
 }
+
+export function runViewBasedReport({
+    query,
+    areaOfConcern,
+}: {
+    query: string;
+    areaOfConcern: string;
+}): Promise<RunReportResponseViewBased> {
+    const requestBody = {
```

Comment:
What are pro and con for `const requestBody: ReportRequestViewBased = {`

To quote Andrew Grove: only the paranoid survive.

Reply:
Ah, but the mental load from paranoia can be costly 👀. Sure, we can add more safeguards here. I realized the error handling needs work too, so I'll fix that as well.